pkg/cri/resource-manager/cache/utils.go

```diff
@@ -198,15 +194,21 @@ func getMemoryCapacity() int64 {
 func cgroupParentToQOS(dir string) corev1.PodQOSClass {
 	var qos corev1.PodQOSClass
 
+	// The parent directory naming scheme depends on the cgroup driver in use.
+	// Thus, rely on substring matching
+	split := strings.Split(strings.TrimPrefix(dir, "/"), "/")
 	switch {
-	case dir == "":
+	case len(split) < 2:
 		qos = corev1.PodQOSClass("")
-	case strings.HasPrefix(dir, cgroupBurstablePrefix):
+	case strings.Index(split[1], "burstable") != -1:
 		qos = corev1.PodQOSBurstable
-	case strings.HasPrefix(dir, cgroupBestEffortPrefix):
+	case strings.Index(split[1], "besteffort") != -1:
 		qos = corev1.PodQOSBestEffort
-	default:
+	case strings.Index(split[1], "guaranteed") != -1:
```

Comment:
this wouldn't work properly. subdirectories are only created to burstable/besteffort. Guaranteed are in the root of kubepods hierarchy.
Also, constants are here: k8s.io/api/core/v1/types.go: "PodQOSBurstable" and so on.

Reply:
Ach, it was my last minute modifications that broke this :/ Thanks for spotting! Now fixed